### PR TITLE
RMET-2666 Local Notifications Plugin - Update pending intents for Android 14

### PR DESCRIPTION
**Pull request by alexgerardojacinto**
## Description
- This PR makes the pending intents that are being used in the plugin immutable.
- Context:  According to the Android 14 Behaviour Changes page (https://developer.android.com/about/versions/14/behavior-changes-14#safer-intents), for an app targeting Android 14, the system restricts it from sending implicit intents to an internal component if the app creates a mutable pending intent (FLAG_MUTABLE) with an intent that does not specify a component or package. When this happens, an exception is thrown.
- We don't need to use mutable pending intents for the Local Notifications plugins. When Android 12 was released and we needed to specify the mutability of Pending Intents (https://developer.android.com/about/versions/12/behavior-changes-12#pending-intent-mutability), we opted for mutable intents, since this was already the default before Android 12. Even more, Android documentation states that “It is strongly recommended to use FLAG_IMMUTABLE when creating a PendingIntent” (https://developer.android.com/reference/android/app/PendingIntent#FLAG_MUTABLE). 
- This PR also replaces the constant value `33554432` with its corresponding constant `PendingIntent.FLAG_MUTABLE`.  In the past, we had to use the actual values because of MABS 7, which targeted API 30. PendingIntent.FLAG_MUTABLE was introduced in API 31.

## Context
https://outsystemsrd.atlassian.net/browse/RMET-2666

## Type of changes

- [ ] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [X] Android
- [ ] iOS
- [ ] JavaScript

## Tests

- Tested on MABS 10 and MABS 9.
- MABS 10 build: https://intranet.outsystems.net/MABS/BuildDetail.aspx?BuildId=3ba36bb449e1d92df641021d23694b516f85d2c7

## Checklist

- [X] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [X] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

File: src/android/notification/Builder.java [hunk 1]
```diff
@@ -327,11 +327,13 @@ private void applyDeleteReceiver(NotificationCompat.Builder builder) {
 
         PendingIntent deleteIntent;
 
-        if(SDK_INT >= 31){
+        if (SDK_INT >= 34) {
```
Comment (hectorjdgdev) [inline src/android/notification/Builder.java:330]: What do you think about to have a local variable that based in the SDK version have PendingIntent.FLAG_IMMUTABLE or 33554432 and that variable is what is passed to  deleteIntent = PendingIntent.getBroadcast(
                    context, reqCode, intent, FLAG_UPDATE_CURRENT | 33554432);

**Review by hectorjdgdev — DISMISSED** [review]

File: src/android/notification/Builder.java (hunk 1 above)
Comment (alexgerardojacinto) [reply src/android/notification/Builder.java:330]: Yes, I actually considered that. We can do that. By the way, this PR is still in draft ahah, it is not ready for review.

Comment (alexgerardojacinto) [reply src/android/notification/Builder.java:330]: I believe that won't work for the else case though. On that case, we're not passing anything in the mutability flag.

Comment (alexgerardojacinto) [reply src/android/notification/Builder.java:330]: I was actually able to refactor and implement your suggestion. Thank you!

Comment (alexgerardojacinto) [reply src/android/notification/Builder.java:330]: Since I applied the requested changes, I am going to resolve this.

**Review by OS-martacarlos — APPROVED** [review]

**Review by nflsilva — APPROVED** [review]

**Pull request merged**